modules/dhcp/dhcp_plugin.rb

```diff
@@ -4,6 +4,7 @@ class Proxy::DhcpPlugin < ::Proxy::Plugin
 
   uses_provider
   default_settings :use_provider => 'dhcp_isc', :server => '127.0.0.1', :subnets => []
+  expose_setting :use_provider
```

Comment:
If you want to expose this manually, I think it'd be easier to do this in the API. You can use `Plugin.uses_provider?` to determine this.

Reply:
Can you elaborate how exactly you want me to do this?

Reply:
If this is your concern I will remove this from this PR. It was just an idea, I have no immediate plan to present this info in Foreman core in any way.

Reply:
I do think it's a valuable debug feature at least, but there's an easier way in `lib/proxy/plugin_initializer.rb` by extending the settings method with:
```ruby
exposed_settings << 'use_provider' if @plugin.uses_provider?
```

Now any module that uses `uses_provider` in the plugin DSL, exposes the `use_provider` setting.

I've found a way to test setting and capabilities for modules and their plugins and I'm preparing a PR with tests and the above fix.